### PR TITLE
Update logging verbosity levels in MTEB

### LINKED ISSUE #1367: Bugs with verbosity
1. When running via Python it seems like the verbosity argument https://github.com/embeddings-benchmark/mteb/blob/d9626abbc5d438024a21e7f21a29d4741bb94188/mteb/evaluation/MTEB.py#L314 does not really matter; Only when running via the CLI it properly adjusts the logging level https://github.com/embeddings-benchmark/mteb/blob/d9626abbc5d438024a21e7f21a29d4741bb94188/mteb/cli.py#L107
2. Here verbosity goes up to 3 https://github.com/embeddings-benchmark/mteb/blob/d9626abbc5d438024a21e7f21a29d4741bb94188/mteb/cli.py#L113 but here it states up to 2 https://github.com/embeddings-benchmark/mteb/blob/d9626abbc5d438024a21e7f21a29d4741bb94188/mteb/evaluation/MTEB.py#L330

### PR DESCRIPTION
**Pull request by YashDThapliyal**
Fixes #1367 

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

### CODE REVIEW

**Comment by YashDThapliyal** [comment]
@Muennighoff @KennethEnevoldsen @isaac-chung 

Updated the verbosity handling in the MTEB evaluation pipeline to ensure consistent logging across the codebase

File: mteb/evaluation/MTEB.py [hunk 1]
```diff
@@ -348,9 +348,23 @@ def run(
...
-        if verbosity < 2:
-            datasets.logging.set_verbosity(40)
-            datasets.logging.disable_progress_bar()
+        # if verbosity < 2:
+        #     datasets.logging.set_verbosity(40)
+        #     datasets.logging.disable_progress_bar()
+
+        #update logging to account for different levels of Verbosity (similar to the command line)
+        
+
```
Comment (isaac-chung) [inline mteb/evaluation/MTEB.py:357]: Can remove.

**Review by isaac-chung — COMMENTED** [review]
Thanks for working on this! I think point#1 in the issue is addressed. Can you please also address point#2 as well? (Update docstrings to include verbosity=3)

**Review by KennethEnevoldsen — APPROVED** [review]
Seems like linting is failing. You can fix this using `make lint`. Otherwise all good

**Comment by isaac-chung** [comment]
Just a reminder that docstrings still need to be updated to match level 3. Not merge-ready.

**Comment by YashDThapliyal** [comment]
@isaac-chung @KennethEnevoldsen 

Thank you for pointing that out! I have now updated the docstring of the run function to include information about verbosity = 3, and I also made the earlier descriptions more descriptive.

Additionally, I ran the linter, and there are no errors in the files I was working on, however it did  highlight an unnecessary list comprehension in mteb/leaderboard/table.py.

Please let me know if you need any further changes or if there are additional issues to address!

**Pull request merged**